### PR TITLE
feat(Tracing): allow custom tracing categories

### PR DESCRIPTION
**Pull request by slohmes**
References #1300, #854

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1306,6 +1306,7 @@ await page.tracing.stop();
 - `options` <[Object]>
   - `path` <[string]> A path to write the trace file to. **required**
   - `screenshots` <[boolean]> captures screenshots in the trace.
+  - `categories` <[Array]<[string]>> specify custom categories to use instead of default.
```
Comment (aslushnikov) [inline docs/api.md:1309]: is there any good resource to link to about tracing categories?

**Review by aslushnikov — APPROVED** [review]
thanks! cc perf gurus @a1ph @pavelfeldman

File: docs/api.md (hunk 1 above)
Comment (paulirish) [reply docs/api.md:1309]: There isn't. Various tools just have a list hardcoded most of the time. You normally find what categories you want by reading devtools/lighthouse/webpagetest source. :)

**Review by paulirish — COMMENTED** [review]
I'm very comfortable with this solution, though I'll offer another take:

1. keep this `categories` option for anyone who wants to take over and control it compeltely
2. add a `jssampling` option to `tracing.start()` for the basic usecase. This will enable hi-res js sampling profiler, a la the `screenshots` option.

**Comment by slohmes** [comment]
@paulirish Sure, that sounds fine to me. For the sampling option, should there be any categories included besides the default categories and `disabled-by-default-v8.cpu_profiler.hires`?

**Comment by paulirish** [comment]
@slohmes no that'd be all.   but you should hold off until we hear back on the API from @aslushnikov or @pavelfeldman

**Comment by aslushnikov** [comment]
@slohmes `jssamping` lgtm

**Comment by aslushnikov** [comment]
This looks good to merge. We can add `jssampling` in a separate PR to close #1300.

**Comment by TomasHubelbauer** [comment]
FYI this doesn't seem to work the way it was intended anymore, see #1300. Setting `categories` explicitly to only a single value (in my case, `disabled-by-default-v8.cpu_profiler`), I still receive events from other (all?) categories.

**Pull request merged**